test/operators/fee.ts

```diff
@@ -0,0 +1,136 @@
+import * as helpers from '../helpers/contract-helpers';
+
+import { expect } from 'chai';
+import { progressTime } from '../helpers/utils';
+
+let ssvNetworkContract: any;
+
+describe('Operator Fee Tests', () => {
+  beforeEach(async () => {
+    ssvNetworkContract = (await helpers.initializeContract()).contract;
+    await helpers.registerOperators(2, 1, helpers.CONFIG.minimalOperatorFee);
+  });
+
+  it('Declare fee success emits OperatorFeeDeclaration event', async () => {
+    await expect(ssvNetworkContract.connect(helpers.DB.owners[2]).declareOperatorFee(1, helpers.CONFIG.minimalOperatorFee +  helpers.CONFIG.minimalOperatorFee / 10))
+      .to.emit(ssvNetworkContract, 'OperatorFeeDeclaration');
+  });
+
+  it('Declare fee success as contract owner', async () => {
+    await ssvNetworkContract.declareOperatorFee(1, helpers.CONFIG.minimalOperatorFee +  helpers.CONFIG.minimalOperatorFee / 10);
```

Comment:
- add gas track with register operator param
- There are many places with this, the basic rule is if it doesnt have emit or revert it should be in gas tracking.

@lior-blox 
- Maybe its best that all write functions have a helper inside contract-helpers.ts file and in this function is built in gas tracking.  
- Also maybe not to write all the dao functions inside contract-helpers.ts, but instead to wrap them with gasTrack inside the test (to not clutter helpers file with functions that will only be tested once).

Reply:
agree